src/event-engine/core/retryPolicy.ts

```diff
@@ -0,0 +1,52 @@
+export class RetryPolicy {
+  static LinearRetryPolicy(configuration: LinearRetryPolicyConfiguration) {
+    return {
+      delay: configuration.delay,
+      maximumRetry: configuration.maximumRetry,
+      shouldRetry(error: any, attempt: number) {
+        if (error?.status?.statusCode === 403) {
+          return false;
+        }
+        return this.maximumRetry > attempt;
+      },
+      getDelay(_: number) {
+        return this.delay * 1000;
+      },
+    };
+  }
+
+  static ExponentialRetryPolicy(configuration: ExponentialRetryPolicyConfiguration) {
+    return {
+      minimumDelay: configuration.minimumDelay,
+      maximumDelay: configuration.maximumDelay,
+      maximumRetry: configuration.maximumRetry,
+
+      shouldRetry(error: any, attempt: number) {
+        if (error?.status?.statusCode === 403) {
+          return false;
+        }
+        return this.maximumRetry > attempt;
+      },
+
+      getDelay(attempt: number) {
+        const calculatedDelay = Math.trunc(Math.pow(2, attempt)) * 1000 + Math.random() * 1000;
```

Comment:
@mohitpubnub what value will be for `attempt` when it called first time? Per specs, it should be 0.

Reply:
It starts from 0. Whenever it enters *Reconnecting state.